frontend/src/pages/news/insideNewsPage.jsx

```diff
@@ -34,14 +37,156 @@ export default function InsideNewsPage() {
+        };
+    }, []);
+
+    useEffect(() => {
+        initialLikeStatusRef.current = initialLikeStatus;
+    }, [initialLikeStatus]);
+
+    useEffect(() => {
+        likeStatusRef.current = likeStatus;
+    }, [likeStatus]);
+
+    const handleLikeStatus = async () => {
+        if (likeStatusRef.current !== initialLikeStatusRef.current) {
+            let interactionType = likeStatusRef.current ? 'like' : 'dislike';
+            saveInteraction(interactionType);
+        }
+    };
+
+    useEffect(() => {
+        const logInteraction = async () => {
+            await saveInteraction('open');
+        };
+        logInteraction();
+    }, []);
+
+    const saveInteraction = async (interactionType) => {
+        if (user) {
+            try {
+                await axios.post(
+                    `${apiEndpoints.url}${apiEndpoints.interactions.saveInteraction}`,
+                    {
+                        user_id: user.id,
+                        blob_id: articleId,
+                        interaction_type: interactionType,
+                    }
+                );
+            } catch (error) {
+                toast.error(`Troubles with saving interaction: ${error}`)
+            }
+        }
+    };
+
+    useEffect(() => {
+        async function getInteractionsCounts() {
+            try {
+                const { data } = await axios.get(`${apiEndpoints.url}${apiEndpoints.interactions.getInteractionsCounts}`, {
+                    params: {
+                        blob_id: articleId,
+                    },
+                });
+                setLikes(data.likes);
+                setViews(data.views);
+            }
+            catch (error) {
+                toast.error(`Troubles with getting interactions count: ${error}`)
+            }
+        }
+        getInteractionsCounts()
+    }, [] );
+
+    useEffect(() => {
+        async function getLikeStatus() {
+            try {
+                const { data } = await axios.get(`${apiEndpoints.url}${apiEndpoints.interactions.getInteractionStatus}`, {
+                    params: {
+                        user_id: user.id,
+                        blob_id: articleId,
+                        interaction_type: 'like'
+                    },
+                });
+                setLikeStatus(data);
+                setInitialLikeStatus(data);
+            } catch (error) {
+                toast.error(`Troubles with getting like status: ${error}`)
+            }
+        }
+
+        if (user && articleId) {
+            getLikeStatus();
+        }
+    }, [user, articleId]);
+
+    const toggleLike = () => {
+        if (user) {
+            setLikeStatus(prev => {
+                const newStatus = !prev;
+                setLikes(count => newStatus ? count + 1 : count - 1);
+                return newStatus;
+            });
+        }
+        else {
+            const notify = () => toast('Sign in to leave your reaction', {
```

Comment:
```suggestion
            const notify = () => toast('Sign in to like this post', {
```

Something like that maybe

We should always assume that our user is a 90-year-old grandma who just got her phone yesterday -> Everything should be as plain and understandable as humanly possible